test-tests/shared/src/test/scala/zio/test/AssertionSpec.scala

```diff
@@ -73,11 +73,12 @@ object AssertionSpec extends ZIOBaseSpec {
       assert(Array(1, 2, 3), equalTo(Array(1, 2, 4)))
     } @@ failure,
     test("equalTo must not have type inference issues") {
-      assert(List(1, 2, 3).filter(_ => false), equalTo(List.empty))
+      assert(List(1, 2, 3).filter(_ => false))(equalTo(List.empty))
     },
-    test("equalTo must fail when comparing two unrelated types") {
-      assert(1, equalTo("abc"))
-    } @@ failure,
+    testM("equalTo must not compile when comparing two unrelated types") {
+      val result = typeCheck("assert(1, equalTo(\"abc\"))")
```

Comment:
Pardon me for commenting here, but shouldn't this use the curried `assert` as well?

Reply:
No pardon necessary, and thank you for the good catch! Fixed now.